apis/openstack-nova/src/main/java/org/jclouds/openstack/nova/v2_0/compute/extensions/NovaSecurityGroupExtension.java

```diff
@@ -135,8 +136,9 @@ public NovaSecurityGroupExtension(NovaApi api,
       }
 
       Set<String> groupNames = instance.getSecurityGroupNames();
-      Set<? extends SecurityGroupInRegion> rawGroups =
-              sgApi.get().list().filter(nameIn(groupNames)).transform(groupToGroupInRegion(region)).toSet();
+       final FluentIterable<org.jclouds.openstack.nova.v2_0.domain.SecurityGroup> allGroups = sgApi.get().list();
+       Set<? extends SecurityGroupInRegion> rawGroups =
+              allGroups.filter(nameIn(groupNames)).transform(groupToGroupInRegion(allGroups, region)).toSet();
```

Comment:
[minor] do you need allGroups variable or the fluentIterable can transform and filter and finally transform again?